shared/filestore/localstore.go

```diff
@@ -188,19 +188,43 @@ func (b *LocalFileBackend) RemoveFile(path string) error {
 	return nil
 }
 
-func (b *LocalFileBackend) ListDirectory(path string) ([]string, error) {
-	var paths []string
-	fileInfos, err := ioutil.ReadDir(filepath.Join(b.directory, path))
+// basePath: path to get to the file but won't be added to the end result
+// path: basePath+path current directory we are looking at
+// maxDepth: parameter to prevent infinite recursion, once this is reached we won't look any further
+func appendRecursively(basePath, path string, maxDepth int) ([]string, error) {
+	results := []string{}
+	dirEntries, err := os.ReadDir(filepath.Join(basePath, path))
 	if err != nil {
 		if os.IsNotExist(err) {
-			return paths, nil
+			return results, nil
 		}
-		return nil, errors.Wrapf(err, "unable to list the directory %s", path)
+		return results, errors.Wrapf(err, "unable to list the directory %s", path)
 	}
-	for _, fileInfo := range fileInfos {
-		paths = append(paths, filepath.Join(path, fileInfo.Name()))
+	for _, dirEntry := range dirEntries {
+		entryName := dirEntry.Name()
+		entryPath := filepath.Join(path, entryName)
+		if entryName == "." || entryName == ".." || entryPath == path {
+			continue
+		}
+		if dirEntry.IsDir() {
```

Comment:
Should we add a test case for this recursion? I Think is not tested in the existing test case and maybe Is something that we should add.